cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -645,6 +646,58 @@ endif()
 
 if (ARROW_AVRO)
   # ----------------------------------------------------------------------
+  # Jansson
+  if("${JANNSON_HOME}" STREQUAL "")
+    set(JANNSON_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/jannson_ep/src/jannson_ep-install")
+    set(JANNSON_HOME "${JANNSON_PREFIX}")
+    set(JANNSON_INCLUDE_DIR "${JANNSON_PREFIX}/include")
+    if (MSVC)
+      set(JANNSON_STATIC_LIB_NAME jannson_static)
+    else()
+      set(JANNSON_STATIC_LIB_NAME jannson)
+    endif()
+    set(JANNSON_STATIC_LIB "${JANNSON_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}${JANNSON_STATIC_LIB_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(JANNSON_SRC_URL "http://www.digip.org/jansson/releases/jansson-${JANNSON_VERSION}.tar.gz")
+
+    if (${UPPERCASE_BUILD_TYPE} EQUAL "RELEASE")
+      if (APPLE)
+        set(JANNSON_CXXFLAGS "CXXFLAGS='-DNDEBUG -O1'")
```

Comment:
Why this differentiation? Maybe copy/paste from snappy? (There we need this because of a compiler bug).

Reply:
oh completely copy paste from snappy